### DISCUSSION TITLE
Default values for User attributes.

### DISCUSSION THREAD

**Discussion by SigmaEmli** [keycloak/keycloak#31158] [Ideas]
Would be nice to be able to set default values to user attributes in keycloak. 
This because, if we want to use SSO and send value/s in JWT token, we can easily manage value/s if a user / admin has not set the value of the user attribute. 
If value is not populated then the attribute in the JWT token will be lost. 
![image](https://github.com/keycloak/keycloak/assets/94954880/2df300c4-222b-442c-9375-ccb4f54c13fa)

**Comment by huyenvu2101** [comment]
Similar needs, we also want to set the default timezone for all users in the realm, then they can manage their own timezone

**Comment by muppeth** [comment]
Was this feature addressed since? It would be great if bunch of attributes could be automatically filled in with data upon user registration. Specially those that are visible only for the admins (eg. fileQuota where default value should be set to 10GB) or attributes that could be derived for example, email = {username}@example.com. 
I've been looking around for an answer to this but so far I have not find any pointers on whether this is possible atm.

**Comment by ssilvert** [comment]
@SigmaEmli @huyenvu2101 @muppeth I also think this is a really good idea.  Please submit an [enhancement request](https://github.com/keycloak/keycloak/issues/new?assignees=&labels=kind%2Fenhancement%2Cstatus%2Ftriage&projects=&template=enhancement.yml) and get everyone to vote on it.

**Reply by muppeth** [reply]
Thanks for reply. I will submit it. But as of now, there is no way to achieve this? I guess custom registration form would be one way, but am I right to assume currently keycloak by default does not provide this?

**Reply by huyenvu2101** [reply]
I submitted a PR for this https://github.com/keycloak/keycloak/pull/39746
Please take a look @muppeth @ssilvert @SigmaEmli

**Comment by ghenadiibatalski** [comment]
link to enhancement request https://github.com/keycloak/keycloak/issues/36160
